From 09554ef0e9c2c56ec790c2b742660227763bf787 Mon Sep 17 00:00:00 2001
From: Wyon Bi <bivvy.bi@rock-chips.com>
Date: Mon, 26 Oct 2020 09:16:53 +0800
Subject: [PATCH] drm/rockchip/rk628: combtxphy: Mark regcache as dirty in
 power on reset

HW registers were reset to default values during power on reset.

Change-Id: I1c78d5f9eb78c80a9a9cfe8c8d106c125fa0cba0
Signed-off-by: Wyon Bi <bivvy.bi@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rk628/rk628_combtxphy.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rk628/rk628_combtxphy.c b/drivers/gpu/drm/rockchip/rk628/rk628_combtxphy.c
index 3b9edaa979a6..bd7531567256 100644
--- a/drivers/gpu/drm/rockchip/rk628/rk628_combtxphy.c
+++ b/drivers/gpu/drm/rockchip/rk628/rk628_combtxphy.c
@@ -210,6 +210,9 @@ static int rk628_combtxphy_power_on(struct phy *phy)
 	reset_control_deassert(combtxphy->rstc);
 	udelay(10);
 
+	regcache_mark_dirty(combtxphy->regmap);
+	regcache_sync(combtxphy->regmap);
+
 	regmap_update_bits(combtxphy->regmap, COMBTXPHY_CON0,
 			   SW_TX_IDLE_MASK | SW_TX_PD_MASK | SW_PD_PLL_MASK,
 			   SW_TX_IDLE(0x3ff) | SW_TX_PD(0x3ff) | SW_PD_PLL);
-- 
2.35.3

